user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md

```diff
@@ -0,0 +1,89 @@
+
+Below you can find the request flow when a trigger comes in. This may help you during your troubleshooting.
+
+```mermaid
+flowchart LR
+client("User") -- "HTTP(S) request" --> a
+subgraph "IIS"
+    a("IIS") --> b("Rewrite Rule")
+end
+subgraph "Extension"
+    b -- HTTP request --> c("UserDefinableApiEndpoint\nDxM")
+end
+subgraph "Core DataMiner"
+    c -- NATS --> d("UserDefinableApiEndpointManager\n(SLNet.exe)") 
+    d -- IPC --> e("SLAutomation.exe")
+end
+```
+
+## Installation issues
+
+The errors in this section can be generated when the UserDefinableApiEndpoint DxM is installed via a DataMiner upgrade package. These will be visible in the upgrade log during the upgrade and also in the `UserDefinableApiEndpoint.txt` log file as mentioned in the [General](#general) section.
+
+### 'Could not find AspNetCore installed on this system'
+
+DataMiner should install ASP.NET Core 5 automatically, but in case this failed or it has been removed, the installer could give this error. To fix this, go to the [download page](https://dotnet.microsoft.com/en-us/download/dotnet/5.0) of .NET 5 and download the latest ASP.NET Core Runtime 5. After that, you can manually rerun the installer which can be found in this  location: `C:\Skyline DataMiner\Tools\ModuleInstallers\DataMiner UserDefinableApiEndpoint 1.X.X.X.msi`.
+
+### 'Found rewrite rule, but has inconsistencies:'
+
+The log line(s) after it will contain the inconsistencies of the IIS rewrite rule. This rewrite rule forwards requests coming in on port 80 or 443 to the UserDefinableApiEndpoint DxM. It should contain the correct port where the DxM is currently running on. The installer will always verify if the port mentioned in the rewrite rule matches the one that is configured in the [app settings of the DxM](xref:UD_APIs_UserDefinableApiEndpoint#configuring-the-dxm).
+
+### 'Port X is in use, this will cause the IIS rewrite rule to be disabled and User Definable APIs to be unusable.'
+
+When no rewrite rule is found in IIS, a new one will be created. When the installer detects that a process is already using the port configured in the app settings of the UserDefinableApiEndpoint DxM, it will create the rule in a disabled state and log this message. This is done to prevent opening up access to an unknown process. The UserDefinableApiEndpoint DxM won't be able to start as long as the port is not free. In this situation you can either move the current process to another port using its configuration, or configure the UserDefinableApiEndpoint DxM to use another port. See the [Kestrel](xref:UD_APIs_UserDefinableApiEndpoint#kestrel) configuration section on the UserDefinableApiEndpoint page for more info. Either way, don't forget to enable the rewrite rule where the port reflects the one configured on the DxM.
```

Comment:
```suggestion
When no rewrite rule is found in IIS, a new one will be created. When the installer detects that a process is already using the port configured in the appsettings file of the UserDefinableApiEndpoint DxM, it will create the rule in a disabled state and log this message. This is done to prevent opening up access to an unknown process. The UserDefinableApiEndpoint DxM won't be able to start as long as the port is not free. In this situation you can either move the current process to another port using its configuration, or configure the UserDefinableApiEndpoint DxM to use another port. See the [Kestrel](xref:UD_APIs_UserDefinableApiEndpoint#kestrel) configuration section on the UserDefinableApiEndpoint page for more info. Either way, don't forget to enable the rewrite rule where the port reflects the one configured on the DxM.
```